### PR TITLE
created the user-CRUD assignment

### PR DESCRIPTION
**Pull request by nikhilatcompro**
Created a vue app that shows the list of user and provides the functionality to execute CRUD operations on the user list.

### CODE REVIEW

File: vuejs/user-display-app/src/App.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    },
+    methods:{
+      showForm(){
+        this.component = 'user-form';
+      },
+
+      showList(){
+        this.component = 'user-list';
+      },
+
```
Comment (PurnimaCompro) [inline vuejs/user-display-app/src/App.vue:35]: Remove extra space

File: vuejs/user-display-app/src/components/Form.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	name: 'user-form',
+	components: {ShowUser},
+	data() {
+		return {
+			id: 0,
+			name: '',
+			email: '',
+			dob: '',
+			editingUser: false,
+		}
```
Comment (PurnimaCompro) [inline vuejs/user-display-app/src/components/Form.vue:31]: alphabetical order

**Comment by PurnimaCompro** [comment]
@nikhilatcompro 

Please indent all files.

File: vuejs/user-display-app/src/App.vue [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    }
+}
+</script>
+
+<style>
+body {
+  background-color: rgb(229, 217, 195);
+}
+
+.add-btn{
```
Comment (chakshuujawa) [inline vuejs/user-display-app/src/App.vue:45]: Please cross-check the styling needed for add-btn. I guess we can eliminate most of them.

File: vuejs/user-display-app/src/components/List.vue [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+</template>
+
+<script>
+import EventBus from "../event-bus.js";
+import ShowUser from "./ShowUser.vue";
+export default {
+    
+    name: "list",
+    data() {
+        return {
```
Comment (chakshuujawa) [inline vuejs/user-display-app/src/components/List.vue:29]: 2 space tab.

File: vuejs/user-display-app/src/components/ShowUser.vue [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+<script>
+import EventBus from '../event-bus';
+export default {
+   name: 'show-user',
+   props: {
+      user: Object   
+    },
+  methods: {
+    deleteUser(id) {
+        EventBus.$emit('delete-user', id);
```
Comment (chakshuujawa) [inline vuejs/user-display-app/src/components/ShowUser.vue:25]: Indentation. 2 space tab
Please check the same for the rest of the file

File: vuejs/user-display-app/src/components/ShowUser.vue [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  background-color: white;
+  height: 20rem;
+  margin: 2rem 0 0 35rem;
+  padding: 4rem;
+  position: relative;
+  text-align: center;
+  width: 15rem;
+}
+
+.back-btn {
```
Comment (chakshuujawa) [inline vuejs/user-display-app/src/components/ShowUser.vue:53]: alphabetical styling

File: vuejs/user-display-app/src/userList.json [hunk 7]
```diff
@@ -0,0 +1,502 @@
+[
```
Comment (chakshuujawa) [inline vuejs/user-display-app/src/userList.json:1]: Please check if this file is needed anymore?

File: vuejs/user-display-app/src/App.vue [hunk 8]
```diff
@@ -0,0 +1,76 @@
+<template>
+  <div>
+    <keep-alive>
```
[the remark this answers is not in the document]
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/App.vue:3]: we are using keep alive tag to maintain the state of each component that changes dynamically.
in our case they are the List and Form components.

File: vuejs/user-display-app/src/App.vue [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    <component @back-clicked='showList' @user-added='showList' 
+    @add-btn-clicked='showForm'  @user-edited='showList' @edit-form='showForm' :is='component'></component>
+    </keep-alive>
+
+  </div>
+</template>
+
+<script>
+import List from './components/list.vue';
```
[the remark this answers is not in the document]
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/App.vue]: done

File: vuejs/user-display-app/src/App.vue (hunk 1 above)
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/App.vue]: done

File: vuejs/user-display-app/src/App.vue [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+body {
+  background-color: rgb(229, 217, 195);
+}
+
+.add-btn{
+  margin:2rem 0 0 25%;
+  background-color: pink;
+  border-radius: 0.5rem;
+  border-style: none;
+  box-sizing: border-box;
```
[the remark this answers is not in the document]
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/App.vue]: done

File: vuejs/user-display-app/src/components/Form.vue [hunk 11]
```diff
@@ -0,0 +1,112 @@
+<template>
+	<div>
```
[the remark this answers is not in the document]
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/components/Form.vue]: done

File: vuejs/user-display-app/src/components/Form.vue (hunk 2 above)
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/components/Form.vue]: done

File: vuejs/user-display-app/src/App.vue (hunk 3 above)
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/App.vue]: done

File: vuejs/user-display-app/src/components/List.vue [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+
+.action-btn:hover {
+  background-color: rgb(39, 171, 127);
+}
+
+.delete-btn:hover {
+  background-color: rgb(235, 36, 92);
+}
+
+td {
```
[the remark this answers is not in the document]
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/components/List.vue]: done

File: vuejs/user-display-app/src/components/ShowUser.vue (hunk 5 above)
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/components/ShowUser.vue:28]: done

File: vuejs/user-display-app/src/components/ShowUser.vue (hunk 6 above)
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/components/ShowUser.vue]: done

File: vuejs/user-display-app/src/userList.json (hunk 7 above)
Comment (nikhilatcompro) [reply vuejs/user-display-app/src/userList.json]: done

**Pull request merged**